File: lib/python/opentoken/src/main/opentoken/keyexchange/key_exchange_exception.py
"""
Copyright (c) Truveta. All rights reserved.
"""


class KeyExchangeException(Exception):
    """
    Exception thrown when key exchange operations fail.

    This exception is used to wrap various cryptographic and I/O exceptions
    that may occur during ECDH key exchange, key derivation, or key management operations.
    """

    def __init__(self, message: str, cause: Exception = None):
        """
        Creates a new KeyExchangeException.

        Args:
            message: The error message.
            cause: The underlying exception that caused this error (optional).
        """
        super().__init__(message)
        self.cause = cause

File: lib/java/opentoken/src/main/java/com/truveta/opentoken/keyexchange/KeyExchangeException.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.keyexchange;

/**
 * Exception thrown when key exchange operations fail.
 * <p>
 * This exception is used to wrap various cryptographic and I/O exceptions
 * that may occur during ECDH key exchange, key derivation, or key management operations.
 */
public class KeyExchangeException extends Exception {
    private static final long serialVersionUID = 1L;
    
    /**
     * Constructs a new KeyExchangeException with the specified detail message.
     *
     * @param message the detail message
     */
    public KeyExchangeException(String message) {
        super(message);
    }
    
    /**
     * Constructs a new KeyExchangeException with the specified detail message and cause.
     *
     * @param message the detail message
     * @param cause the cause of the exception
     */
    public KeyExchangeException(String message, Throwable cause) {
        super(message, cause);
    }
}
